src/test/java/com/snowflake/kafka/connector/records/ConverterTest.java

```diff
@@ -258,6 +259,19 @@ public void testConnectJsonConverter_MapBigDecimalExceedsMaxPrecision() throws J
     //assert expected.toString().equals(result.toString());
   }
 
+  @Test
+  public void testConnectSimpleHeaderConverter_MapDateAndOtherTypes() throws JsonProcessingException {
+    SimpleHeaderConverter headerConverter = new SimpleHeaderConverter();
+    String rawHeader = "{\"f1\": \"1970-03-22\", \"f2\": true}";
+    SchemaAndValue schemaAndValue = headerConverter.toConnectHeader("test", "h1", rawHeader.getBytes(StandardCharsets.US_ASCII));
+    JsonNode result = RecordService.convertToJson(schemaAndValue.schema(), schemaAndValue.value());
```

Comment:
Is this the right place to add a RecordService test? every other test in this file seems to be testing custom avro/json converters, while this is using a stock simple header converter.

Reply:
The converters in this test aren't all custom; there are also cases including the `JsonConverter`, which comes out of the box with Connect. And the `SimpleHeaderConverter` is, at this point, anything but simple; the parsing logic in it is at least as complex as the stuff in the `JsonConverter`.